blockchain.go

```diff
@@ -291,6 +296,31 @@ func (b *Blockchain) GetBlockLocatorHashes() []*chainhash.Hash {
 	return ret
 }
 
+// Returns last header before reorg point
+func (b *Blockchain) GetLastGoodHeader(bestHeader, prevBestHeader StoredHeader) (*StoredHeader, error) {
+	majority, err := b.db.GetPreviousHeader(bestHeader.header)
+	if err != nil {
+		return nil, err
+	}
+	minority := prevBestHeader
+	for {
```

Comment:
I think this needs a hard upper bound, even if it "should" always finish, just to be defensive. Even a really large number is better than nothing. Otherwise there's some non-zero chance this could end up going forever and eating the CPU.